lib/cretonne/src/ir/frontend.rs

```diff
@@ -0,0 +1,764 @@
+    pub fn declare_var(&mut self, var: Variable, ty: Type) {
+        *self.builder.types.ensure(var) = ty;
+    }
+
+    /// Returns the Cretonne IL value corresponding to the utilization at the current program
+    /// position of a previously defined user variable.
+    pub fn use_var(&mut self, var: Variable) -> Value {
+        let ty = match self.builder.types.get(var) {
+            Some(&ty) => ty,
+            None => panic!("this variable is used but its type has not been declared"),
+        };
+        self.builder
+            .ssa
+            .use_var(&mut self.func.dfg, var, ty, self.position.basic_block)
+    }
+
+    /// Register a new definition of a user variable. Panics if the type of the value is not the
+    /// same as the type registered for the variable.
+    pub fn def_var(&mut self, var: Variable, val: Value) {
+        debug_assert!(self.func.dfg.value_type(val) == self.builder.types[var],
+                      "the type of the value is not the type registered for the variable");
+        self.builder
+            .ssa
+            .def_var(var, val, self.position.basic_block);
+    }
+
+    /// Returns the value corresponding to the `i`-th argument of the function as defined by
+    /// the function signature. Panics if `i` is out of bounds or if called before the first call
+    /// to `switch_to_block`.
+    pub fn arg_value(&self, i: usize) -> Value {
+        debug_assert!(!self.pristine, "you have to call switch_to_block first.");
+        self.builder.function_args_values[i]
+    }
+
+    /// Creates a jump table in the function, to be used by `br_table` instructions.
+    pub fn create_jump_table(&mut self) -> JumpTable {
+        self.func.jump_tables.push(JumpTableData::new())
+    }
+
+    /// Inserts an entry in a previously declared jump table.
+    pub fn insert_jump_table_entry(&mut self, jt: JumpTable, index: usize, ebb: Ebb) {
+        self.func.jump_tables[jt].set_entry(index, ebb);
+    }
+
+    /// Creates a stack slot in the function, to be used by `stack_load`, `stack_store` and
+    /// `stack_addr` instructions.
+    pub fn create_stack_slot(&mut self, data: StackSlotData) -> StackSlot {
+        self.func.stack_slots.push(data)
+    }
+
+    /// Adds a signature which can later be used to declare an external function import.
+    pub fn import_signature(&mut self, signature: Signature) -> SigRef {
+        self.func.dfg.signatures.push(signature)
+    }
+
+    /// Declare an external function import.
+    pub fn import_function(&mut self, data: ExtFuncData) -> FuncRef {
+        self.func.dfg.ext_funcs.push(data)
+    }
+
+
+
+    /// Returns an object with the [`InstBuilder`](../trait.InstBuilder.html) trait that allows to
+    /// conveniently append an instruction to the current `Ebb` being built.
+    pub fn ins<'short>(&'short mut self) -> FuncInstBuilder<'short, 'a, Variable> {
+        let ebb = self.position.ebb;
+        FuncInstBuilder::new(self, ebb)
+    }
```

Comment:
I wish I could do that but the signature of the build function is `build(self, ...)` and not `build(&mut self)`. Because `self` is consumed I can't make `self` the whole `FunctionBuilder`, and as a consequence we're stuck with the quirky `builder.ins()....` syntax.